lib/services/role.go

```diff
@@ -1979,6 +1995,36 @@ func (set RoleSet) EnhancedRecordingSet() map[string]bool {
 	return m
 }
 
+func (set RoleSet) CreateHostUser(s types.Server) bool {
```

Comment:
Also, please add tests for CreateHostUser and HostGroups method that play out various scenarios when you have multiple roles.